pkg/cmd/cli/cmd/newapp.go

```diff
@@ -511,6 +522,24 @@ func setLabels(labels map[string]string, result *newcmd.AppResult, ignoreFailure
 	return nil
 }
 
+func hasAppLabel(result *newcmd.AppResult) (bool, error) {
```

Comment:
yeah we should probably just do what you suggested... copy the objects, call set labels (with a flag to error on conflict), if it doesn't fail, use the new objects, it it fails, use the old objects.  Was hoping to avoid the copy.

As for why AddObjectLabels does what it does/doesn't, i don't know.  it does seem fragile.